File: 2018/python/day14.py
input = [3, 7]
elves = [0, 1]
pat = [7,6,5,0,7,1]

while True:
  newValue = input[elves[0]] + input[elves[1]]
  if newValue // 10 != 0: # two digit value
    input.append(newValue // 10) # first digit
    if pat == input[len(input)-len(pat):]:
      break
  input.append(newValue % 10) # second (or only) digit
  if pat == input[len(input)-len(pat):]:
    break

  elves[0] = (elves[0]+1+input[elves[0]]) % len(input)
  elves[1] = (elves[1]+1+input[elves[1]]) % len(input)

print("Ten recipes after sequence: {}".format(input[765071:765071+10]))
print("#recipes before sequence: {}".format(len(input)-len(pat)))

File: go/src/2018/day14/day14.go
package main

import (
	"fmt"
	"reflect"
)

func findLast(s []int, p []int) bool {
	if len(s) < len(p) {
		return false
	}
	if reflect.DeepEqual(s[len(s)-len(p):], p) {
		return true
	}
	return false
}

func main() {
	input := []int{3, 7}
	elves := [2]int{0, 1}
	pattern := []int{7, 6, 5, 0, 7, 1}

	for {
		newValue := input[elves[0]] + input[elves[1]]
		if newValue/10 != 0 {
			input = append(input, newValue/10)
			if findLast(input, pattern) {
				break
			}
		}
		input = append(input, newValue%10)
		if findLast(input, pattern) {
			break
		}
		elves[0] = (elves[0] + 1 + input[elves[0]]) % len(input)
		elves[1] = (elves[1] + 1 + input[elves[1]]) % len(input)
	}
	fmt.Printf("Ten recipes after sequence: %v\n", input[765071:765071+10])
	fmt.Printf("#recipes before sequence: %d", len(input)-len(pattern))
}
